Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs

```diff
@@ -1821,3 +1807,31 @@ fetchStageFromRideStatus activeRide =
     COMPLETED -> ST.RideCompleted
     CANCELLED -> ST.HomeScreen
     _ -> ST.HomeScreen
+
+openNav :: ST.HomeScreenState -> Number -> Number -> Effect Unit
+openNav state lat lon = do 
+  case state.props.currentStage of 
+    ST.RideAccepted -> do
+      let navType = case state.data.vehicleType of
+                      "AUTO_RICKSHAW" -> state.data.cityConfig.travelMode.pickup.autoRickshaw
+                      "TAXI" -> state.data.cityConfig.travelMode.pickup.taxi
+                      "BIKE" -> state.data.cityConfig.travelMode.pickup.bike
+                      "TAXI_PLUS" -> state.data.cityConfig.travelMode.pickup.taxiPlus
+                      "SEDAN" -> state.data.cityConfig.travelMode.pickup.sedan
+                      "SUV" -> state.data.cityConfig.travelMode.pickup.suv
+                      "HATCHBACK" -> state.data.cityConfig.travelMode.pickup.hatchback
+                      "SUV_PLUS" -> state.data.cityConfig.travelMode.pickup.suvPlus
+                      _ -> "DRIVE"
+      void $ pure $ openNavigation lat lon navType
+    _ -> do
+      let navType = case state.data.vehicleType of
+                      "AUTO_RICKSHAW" -> state.data.cityConfig.travelMode.dropOff.autoRickshaw
+                      "TAXI" -> state.data.cityConfig.travelMode.dropOff.taxi
+                      "BIKE" -> state.data.cityConfig.travelMode.dropOff.bike
+                      "TAXI_PLUS" -> state.data.cityConfig.travelMode.dropOff.taxiPlus
+                      "SEDAN" -> state.data.cityConfig.travelMode.dropOff.sedan
+                      "SUV" -> state.data.cityConfig.travelMode.dropOff.suv
+                      "HATCHBACK" -> state.data.cityConfig.travelMode.dropOff.hatchback
+                      "SUV_PLUS" -> state.data.cityConfig.travelMode.dropOff.suvPlus
+                      _ -> "DRIVE"
```

Comment:
we can store the `state.data.cityConfig.travelmode.pickup` and `state.data.cityConfig.travelmode.dropOff` in one variable and use that in all places